kratos/mpi/tests/test_mpi_serializer.py

```diff
@@ -0,0 +1,103 @@
+import KratosMultiphysics.mpi as KratosMPI
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+from KratosMultiphysics.mpi import distributed_import_model_part_utility
+import KratosMultiphysics.KratosUnittest as UnitTest
+import sys
+
+# Use cPickle on Python 2.7 (Note that only the cPickle module is supported on Python 2.7)
+# Source: https://pybind11.readthedocs.io/en/stable/advanced/classes.html
+pickle_message = ""
+try:
+    import cPickle as pickle
+    have_pickle_module = True
+except ImportError:
+    if sys.version_info > (3, 0):
+        try:
+            import pickle
+            have_pickle_module = True
+        except ImportError:
+            have_pickle_module = False
+            pickle_message = "No pickle module found"
+    else:
+        have_pickle_module = False
+        pickle_message = "No valid pickle module found"
+
+def executeComputeArea_Task(pickled_model):
+    # Unpickling model
+    serialized_model = pickle.loads(pickled_model)
+
+    # Unserializing model
+    deserialized_model = KratosMultiphysics.Model()
+    serialized_model.Load("ModelSerialization", deserialized_model)
+
+    model_part = deserialized_model.GetModelPart("ModelPart").GetRootModelPart()
+    ## Construct and execute the Parallel fill communicator
+    ParallelFillCommunicator = KratosMPI.ParallelFillCommunicator(model_part)
+    ParallelFillCommunicator.Execute()
+    communicator = model_part.GetCommunicator()
+
+    # Computing local areas
+    for node in model_part.Nodes:
+        node.SetValue(KratosMultiphysics.NODAL_AREA, 0.0)
+
+    for elem in model_part.Elements:
+        for node in elem.GetNodes():
+            current_nodal_area = node.GetValue(KratosMultiphysics.NODAL_AREA)
+            current_nodal_area += 1/3*elem.GetGeometry().Area()
+            node.SetValue(KratosMultiphysics.NODAL_AREA, current_nodal_area)
+
+    # Assembling nodal values
+    communicator.AssembleNonHistoricalData(KratosMultiphysics.NODAL_AREA)
+
+    # Computing sum of total area to check results.
+    local_sum = sum(node.GetValue(KratosMultiphysics.NODAL_AREA) for node in model_part.Nodes if node.GetSolutionStepValue(KratosMultiphysics.PARTITION_INDEX) == communicator.GetDataCommunicator().Rank())
+    total_sum = communicator.GetDataCommunicator().SumAll(local_sum)
+
+    return total_sum
+
+class TestMPISerializer(UnitTest.TestCase):
+
+    def setUp(self):
+        communicator = KratosMultiphysics.DataCommunicator.GetDefault()
+        rank = communicator.Rank()
+
+        current_model = KratosMultiphysics.Model()
+        model_part_to_read = current_model.CreateModelPart("ModelPart")
+        model_part_to_read.AddNodalSolutionStepVariable(KratosMultiphysics.PARTITION_INDEX)
+
+        importer_settings = KratosMultiphysics.Parameters("""{
+            "model_import_settings": {
+                "input_type": "mdpa",
+                "input_filename": "test_mpi_serializer",
+                "partition_in_memory" : false
```

Comment:
why not?
Then you don't have to remove the partitioned files afterwards

I use it in my tests and works fine

Reply:
Aside of the fact that it is much faster.